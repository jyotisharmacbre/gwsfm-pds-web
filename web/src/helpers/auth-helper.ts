// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

