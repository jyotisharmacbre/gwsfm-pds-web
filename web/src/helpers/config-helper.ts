const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;