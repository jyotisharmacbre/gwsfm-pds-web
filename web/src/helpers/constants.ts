export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
