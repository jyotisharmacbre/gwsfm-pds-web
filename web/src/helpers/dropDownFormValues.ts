export const projectStatusData: object[] = [
  { label: '1', value: '1' },
  { label: '2', value: '2' },
  { label: '3', value: '3' }
];

export const engagementData: object[] = [
  { label: 'Type 1', value: '1' },
  { label: 'Type 2', value: '2' },
  { label: 'Type 3', value: '3' }
];

export const typesOfEngagement: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Fixed Price',
    value: 'Fixed Price'
  },
  {
    label: 'Time and Materials',
    value: 'Time and Materials'
  },
  {
    label: 'Risk and Reward',
    value: 'Risk and Reward'
  },
  {
    label: 'Cost Plus',
    value: 'Cost Plus'
  }
];

export const contractType: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'CT 1',
    value: 'CT 1'
  },
  {
    label: 'CT 2',
    value: 'CT 2'
  }
];

export const currency: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AED (UAE Dirham)',
    value: 'AED'
  },
  {
    label: 'AFN (Afghani)',
    value: 'AFN'
  },
  {
    label: 'ALL (Lek)',
    value: 'ALL'
  },
  {
    label: 'AMD (Dram)',
    value: 'AMD'
  },
  {
    label: 'ANG (Dutch Antilles Guilder)',
    value: 'ANG'
  },
  {
    label: 'AOA (Kwanza)',
    value: 'AOA'
  },
  {
    label: 'ARS (Peso)',
    value: 'ARS'
  }
];

export const projectStatus: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Initial Customer Inquiry',
    value: 'Initial Customer Inquiry'
  },
  {
    label: 'J&A',
    value: 'J&A'
  },
  {
    label: 'Archived',
    value: 'Archived'
  },
  {
    label: 'Lost',
    value: 'Lost'
  },
  {
    label: 'Order Received',
    value: 'Order Received'
  },
  {
    label: 'CPP',
    value: 'CPP'
  },
  {
    label: 'Complete',
    value: 'Complete'
  }
];

export const assetsWorkedOn: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AS 1',
    value: 'AS 1'
  },
  {
    label: 'AS 2',
    value: 'AS 2'
  }
];

export const enquiryTypeData: object[] = [
  { label: 'Written', value: '1' },
  { label: 'Verbal', value: '2' },
  { label: 'Full tender enquiry with specifications & orgs', value: '3' },
  { label: 'Other', value: '4' }
];
