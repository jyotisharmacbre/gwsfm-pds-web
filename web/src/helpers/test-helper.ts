import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
