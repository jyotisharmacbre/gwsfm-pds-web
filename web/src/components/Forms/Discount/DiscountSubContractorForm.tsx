import { faPlusCircle, faTrash } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Field } from 'redux-form';
import { Validate } from '../../../helpers/fieldValidations';
import { newSubContractorDiscount } from '../../../store/DiscountForm/InitialState';
import ValidatedNumericInput from '../../NumericInput/index';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import PdsFormTextArea from '../../PdsFormHandlers/PdsFormTextArea';
import { confirmAlert } from '../../Popup/CustomModalPopup';
import { restrictMinusAndAllowDecimal } from '../../../helpers/utility-helper';


interface Props {
    fields: any;
    currencySymbol: string;
    intl: any;
}
const DiscountSubContractorForm: React.FC<Props> = (props: Props) => {
    const { fields, intl } = props;
    console.log(intl);
    return (
        <div>
            {fields.map((member, index) => (
                <div className="row" key={index} data-test="discount-sub-contractor-form">
                    <div className="col-lg-9">
                        <div className="forms_wrap">

                            {fields.length > 1 ? (

                                <button
                                    data-test="deleteDiscountSubContractor"
                                    className="delete_text"
                                    onClick={() =>
                                        confirmAlert({
                                            intl: intl,
                                            titleKey: 'TITLE_CONFIRMATION',
                                            contentKey: 'MESSAGE_DISCOUNT_SUBCONTRACTOR_DELETE_ACTIVITY',
                                            handleConfirm: () => fields.remove(index)
                                        })
                                    }
                                >
                                    <FormattedMessage id="BUTTON_DELETE" />
                                    <FontAwesomeIcon className="" icon={faTrash} />
                                </button>
                            ) : null}
                            <div className="discount-subcontractor-section">
                                <Field
                                    name={`${member}.supplierName`}
                                    type="text"
                                    component={PdsFormInput}
                                    validate={[Validate.maxLength(1000)]}
                                    messageKey="MESSAGE_SUPPLIER_NAME"
                                    labelKey="LABEL_SUPPLIER"
                                    placeholderKey="PLACEHOLDER_ENTER_SUPPLIER_NAME"
                                />
                                <Field
                                    name={`${member}.supplierState`}
                                    type="text"
                                    component={PdsFormInput}
                                    validate={[Validate.maxLength(1000)]}
                                    messageKey="MESSAGE_STATE_DETAILS_NAME"
                                    labelKey="LABEL_STATE_DETAILS"
                                    placeholderKey="PLACEHOLDER_ENTER_STATE_DETAILS"
                                />
                                <Field
                                    name={`${member}.supplierTotalDiscount`}
                                    type="number"
                                    className="width-120 pl-20"
                                    component={PdsFormInput}
                                    validate={[Validate.maxLength(15)]}
                                    messageKey="MESSAGE_TOTAL_DISCOUNT"
                                    labelKey="LABEL_TOTAL_DISCOUNT"
                                    currency={props.currencySymbol}
                                    divPosition="relative"
                                    normalize={restrictMinusAndAllowDecimal}

                                />
                                <Field
                                    name={`${member}.supplierComments`}
                                    rows={7}
                                    type="textarea"
                                    component={PdsFormTextArea}
                                    validate={[Validate.maxLength(5000)]}
                                    labelKey="LABEL_COMMENTS"
                                    placeholderKey="PLACEHOLDER_ADDITIONAL_COMMENTS"
                                />
                            </div>
                        </div>
                    </div>
                </div>

            ))}
            <div className="newDiscountSubcontractor_btn">
                <button
                    data-test="addSubContractorDiscount"
                    name="addSubContractorDiscount"
                    type="button"
                    hidden={fields.length > 4}
                    className="active"
                    onClick={() => { fields.push({ ...newSubContractorDiscount }); console.log(fields.length); }}
                >
                    <FontAwesomeIcon className="" icon={faPlusCircle} />
                    <FormattedMessage id="BUTTON_NEW_SUBCONTRACTOR_DISCOUNT" />
                </button>
            </div>
        </div>
    );
};

export default DiscountSubContractorForm;


