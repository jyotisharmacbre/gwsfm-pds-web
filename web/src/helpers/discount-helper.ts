import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;