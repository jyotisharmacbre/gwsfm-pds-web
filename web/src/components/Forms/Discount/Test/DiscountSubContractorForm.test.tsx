import { shallow } from 'enzyme';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { findByTestAtrr } from '../../../../helpers/test-helper';
import translations from '../../../../Translations/translation';
import DiscountSubContractorForm from '../DiscountSubContractorForm';
import * as testData from '../Test/DiscountSubContractorFormTestData';

let wrapper: any;

let props = {
    fields: testData.getSubContractorDiscounts(2),
    currencySymbol: testData.currencySymbol,
    intl: (new IntlProvider({ locale: "en", messages: translations['en'].messages })).state.intl
};

const shallowWrapper = (props) => {
    wrapper = shallow(
        <DiscountSubContractorForm
            {...props}
        />
    );
}

describe('Discount Sub Contractor Form tests', () => {
    beforeEach(() => {
        shallowWrapper(props);
    });

    it('Defines the component', () => {
        expect(wrapper).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('renders the discount SubContractor individual section with no errors', () => {
        expect(findByTestAtrr(wrapper, 'discount-sub-contractor-form').length).toEqual(2);
    });

    it('should renders the delete button when multiple discount', () => {
        expect(findByTestAtrr(wrapper, 'deleteDiscountSubContractor').length).toEqual(2);
    });

    it('should not renders the delete button when only one sub contractor discount', () => {
        props.fields = testData.getSubContractorDiscounts(1);
        shallowWrapper(props);
        expect(findByTestAtrr(wrapper, 'deleteDiscountSubContractor').length).toEqual(0);
    });

    it('test the add addSubContractorDiscount click event', () => {
        let addSubContractorDiscount = findByTestAtrr(wrapper, 'addSubContractorDiscount');
        addSubContractorDiscount.simulate('click');
        wrapper.update();
    });

    it('should disable the new activity button when subcontractor discount length is greater than 5', () => {
        props.fields = testData.getSubContractorDiscounts(5);
        shallowWrapper(props);
        let addSubContractorDiscount = findByTestAtrr(wrapper, 'addSubContractorDiscount');
        wrapper.update();
        expect(addSubContractorDiscount.prop('disabled')).toBe(true);
    });
    it('should show the popup for confirm on delete button click on subcontractor discount', () => {
        let deleteSubContractorDiscountButtons = findByTestAtrr(wrapper, 'deleteDiscountSubContractor');
        deleteSubContractorDiscountButtons.last().simulate('click');
        wrapper.update();
        let doc: any = (document.getElementById('react-confirm-alert'));
        expect(doc.innerHTML).not.toBeNull();
        expect(doc.childElementCount).toBe(1);
    });
});
