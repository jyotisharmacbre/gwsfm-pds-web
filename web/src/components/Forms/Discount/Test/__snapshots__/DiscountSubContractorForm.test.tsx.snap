// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discount Sub Contractor Form tests should match the snapshot 1`] = `ShallowWrapper {}`;
